packages/runtime/test/modules/DeciderModule.test.ts

```diff
@@ -1,126 +1,3228 @@
-import { RelationshipTemplateContent, Request } from "@nmshd/content";
-import { CoreDate } from "@nmshd/core-types";
+import { NodeLoggerFactory } from "@js-soft/node-logger";
+import { LocalAttributeDeletionStatus } from "@nmshd/consumption";
 import {
+    CreateAttributeAcceptResponseItemJSON,
+    DeleteAttributeAcceptResponseItemJSON,
+    FreeTextAcceptResponseItemJSON,
+    GivenName,
+    GivenNameJSON,
+    IdentityAttribute,
+    IdentityFileReferenceJSON,
+    ProposeAttributeAcceptResponseItemJSON,
+    ProprietaryFileReferenceJSON,
+    ProprietaryStringJSON,
+    ReadAttributeAcceptResponseItemJSON,
+    RegisterAttributeListenerAcceptResponseItemJSON,
+    RejectResponseItemJSON,
+    RelationshipAttribute,
+    RelationshipAttributeConfidentiality,
+    RelationshipTemplateContent,
+    Request,
+    ResponseItemGroupJSON,
+    ResponseResult,
+    ShareAttributeAcceptResponseItemJSON
+} from "@nmshd/content";
+import { CoreAddress, CoreDate } from "@nmshd/core-types";
+import {
+    AcceptResponseConfig,
+    AuthenticationRequestItemConfig,
+    ConsentRequestItemConfig,
+    CreateAttributeRequestItemConfig,
+    DeleteAttributeAcceptResponseConfig,
+    DeleteAttributeRequestItemConfig,
+    FreeTextAcceptResponseConfig,
+    FreeTextRequestItemConfig,
+    GeneralRequestConfig,
+    ProposeAttributeRequestItemConfig,
+    ProposeAttributeWithNewAttributeAcceptResponseConfig,
+    ReadAttributeRequestItemConfig,
+    ReadAttributeWithNewAttributeAcceptResponseConfig,
+    RegisterAttributeListenerRequestItemConfig,
+    RejectResponseConfig,
+    ShareAttributeRequestItemConfig
+} from "src/modules/decide";
+import {
+    DeciderModule,
+    DeciderModuleConfigurationOverwrite,
     IncomingRequestStatusChangedEvent,
     LocalRequestStatus,
     MessageProcessedEvent,
     MessageProcessedResult,
     RelationshipTemplateProcessedEvent,
     RelationshipTemplateProcessedResult
 } from "../../src";
-import { RuntimeServiceProvider, TestRequestItem, TestRuntimeServices, establishRelationship, exchangeMessage } from "../lib";
+import { RuntimeServiceProvider, TestRequestItem, TestRuntimeServices, establishRelationship, exchangeMessage, executeFullCreateAndShareRepositoryAttributeFlow } from "../lib";
 
 const runtimeServiceProvider = new RuntimeServiceProvider();
 
-let sender: TestRuntimeServices;
-let recipient: TestRuntimeServices;
+afterAll(async () => await runtimeServiceProvider.stop());
 
-beforeAll(async () => {
-    const runtimeServices = await runtimeServiceProvider.launch(2, { enableDeciderModule: true });
+describe("DeciderModule", () => {
+    describe("Unit tests", () => {
```

Comment:
you could extract the unit tests to a file called `DeciderModule.unit.test.ts` to make the test file less than 12371298321381 lines :D

Reply:
I'm not a big fan of having

1. that deeply nested describes
2. describes that explain what kind of testing is done

Reply:
> 12371298321381 lines

Bye bye 😢

Reply:
:D